jccli/jc_api_v1.py

```diff
@@ -180,3 +181,68 @@ def set_user(self, username, attributes):
             body=Systemuserput(**attributes)
         )
         return api_response.to_dict()
+
+    def search_systems(self, filter={}):
+        """
+        Search for users on JumpCloud.
+
+        :param filter: (dict) an object used to filter search results for various fields. E.g.: `{"firstname": "David"}`
+        :return:
+        """
+        query_filter = {'and': []}
+        for field, value in filter.items():
+            query_filter['and'].append({field: value})
+        if not filter:
+            query_filter = None
+
+        try:
+            api_response = self.search_api.search_systems_post(
+                content_type='application/json',
+                accept='application/json',
+                body={
+                    'filter': query_filter
+                }
+            )
+            systems = [system.to_dict() for system in api_response.results]
+            return systems
+        except ApiException as error:
+            raise "Exception when calling SystemusersApi->systems_search: %s\n" % error
```

Comment:
Could use a unit test for these errors to since I don't think they're being exercised now or they should have revealed a Python trying to raise a bare string.

Reply:
I went with just wrapping the string with an `Exception()`. Kind of a slapped-on solution, although for the moment I can't think of what the better way to handle it is.

Reply:
Okay, if it would be useful to have the stack of the underlying jcapiv2.ApiException whe one of these wrapping exceptions is raised then using the chained `from` as in `raise Exception(message) from error` will preserve the underlying stack trace. Generally it's useful to always chain the source exception unless you are trying to explicitly hide/simplify the source away from the caller.

Reply:
Done